src/main/scala/edu/berkeley/ce/rockslicing/LoadBalancer.scala

```diff
@@ -0,0 +1,323 @@
+package edu.berkeley.ce.rockslicing
+
+import breeze.linalg
+import breeze.linalg.{DenseVector, DenseMatrix}
+import scala.annotation.tailrec
+
+/** 
+  * Manages initial partitioning of rock volume to maintain load balance among partitions
+  */
+object LoadBalancer {
+
+  /**
+    * Produce processor joints that cut initial rock volume into approximately equal volumes.
+    * These joints will be used to seed the initial Block RDD such that the volume or rock
+    * provided to each process is approximately equal.
+    * @param rockVolume Initial block defining the entire rock volume of interest
+    * @param numSeedJoints Number of joints to process before initiating parallel block cutting.
+    *                      Should be one less than number of cores used.
+    * @return Set of processor joints that will be used to seed the initial RDD
+    */
+  def generateSeedJoints(rockVolume: Block, numSeedJoints: Integer):
+                         Seq[Joint] = {
+    // Calculate bounding box
+    val vertices = rockVolume.findVertices.values.flatten
+    val x_max = (vertices.map{ case vertex => vertex._1}).max
+    val x_min = (vertices.map{ case vertex => vertex._1}).min
+    val y_max = (vertices.map{ case vertex => vertex._2}).max
+    val y_min = (vertices.map{ case vertex => vertex._2}).min
+    val z_max = (vertices.map{ case vertex => vertex._3}).max
+    val z_min = (vertices.map{ case vertex => vertex._3}).min
+    val boundingBox = (x_min, y_min, z_min, x_max, y_max, z_max)
+
+    // Diagonal vector from lower left to upper right corner of bounding box
+    val diagonalVector = breeze.linalg.normalize(DenseVector[Double](boundingBox._4 - boundingBox._1,
+                                                                     boundingBox._5 - boundingBox._2,
+                                                                     boundingBox._6 - boundingBox._3))
+    val volumePerPart = rockVolume.volume/(numSeedJoints + 1)
+    val centroid = rockVolume.centroid
+    val centroidVolume = Block(centroid, rockVolume.updateFaces(centroid))
+    val seedJoints =
+      findProcessorJoints(Seq[Joint](), (centroidVolume.centerX, centroidVolume.centerY, centroidVolume.centerZ),
```

Comment:
For the first argument, how about `Seq.empty[Joint]`. It's just a bit more explicit, but not really a big deal.